fastlane/lanes/build.rb

```diff
@@ -0,0 +1,427 @@
+
+    comment_body = "You can test the <strong>Jetpack</strong> changes on this Pull Request by downloading it from AppCenter <a href='#{install_url}'>here</a> with build number: <code>#{build_number}</code>. IPA is available <a href='#{download_url}'>here</a>. If you need access to this, you can ask a maintainer to add you."
+
+    comment_on_pr(
+      project: 'wordpress-mobile/wordpress-ios',
+      pr_number: Integer(ENV['BUILDKITE_PULL_REQUEST']),
+      reuse_identifier: 'jetpack-installable-build-link',
+      body: comment_body
+    )
+  end
+
+  # Builds the "WordPress Internal" scheme and upload it to AppCenter for internal testing
+  #
+  # @option [Boolean] skip_confirm (default: false) If true, avoids any interactive prompt
+  # @option [Boolean] skip_prechecks (default: false) If true, don't run the ios_build_prechecks and ios_build_preflight
+  #
+  # @called_by build_and_upload_app_center
+  #
+  desc 'Builds and uploads for distribution'
+  lane :build_and_upload_internal do |options|
+    ios_build_prechecks(skip_confirm: options[:skip_confirm], internal: true) unless options[:skip_prechecks]
+    ios_build_preflight unless options[:skip_prechecks]
+
+    sentry_check_cli_installed
+
+    internal_code_signing
+
+    gym(
+      scheme: 'WordPress Internal',
+      workspace: WORKSPACE_PATH,
+      export_method: 'enterprise',
+      clean: true,
+      output_directory: BUILD_PRODUCTS_PATH,
+      output_name: 'WordPress Internal',
+      derived_data_path: DERIVED_DATA_PATH,
+      export_team_id: get_required_env('INT_EXPORT_TEAM_ID'),
+      export_options: { method: 'enterprise' }
+    )
+
+    appcenter_upload(
+      api_token: ENV['APPCENTER_API_TOKEN'],
+      owner_name: 'automattic',
+      owner_type: 'organization',
+      app_name: 'WP-Internal',
+      file: lane_context[SharedValues::IPA_OUTPUT_PATH],
+      dsym: lane_context[SharedValues::DSYM_OUTPUT_PATH],
+      notify_testers: false
+    )
+
+    sentry_upload_dsym(
+      auth_token: get_required_env('SENTRY_AUTH_TOKEN'),
+      org_slug: 'a8c',
+      project_slug: 'wordpress-ios',
+      dsym_path: lane_context[SharedValues::DSYM_OUTPUT_PATH]
+    )
+  end
+
+  # Builds the WordPress app and uploads it to TestFlight, for beta-testing or final release
+  #
+  # @option [Boolean] skip_confirm (default: false) If true, avoids any interactive prompt
+  # @option [Boolean] skip_prechecks (default: false) If true, don't run the ios_build_prechecks and ios_build_preflight
+  # @option [Boolean] create_gh_release If tru, creates a GitHub Release draft after the upload, with zipped xcarchive as artefact
+  # @option [Boolean] beta_release If true, the GitHub release will be marked as being a pre-release
+  #
+  # @called_by build_and_upload_app_store_connect
+  #
+  desc 'Builds and uploads WordPress to TestFlight for distribution'
+  lane :build_and_upload_itc do |options|
```

Comment:
Note that while doing that big spring cleaning here, that allowed me to notice that `build_and_upload_itc` lane is in fact:
 - Only called by the `build_and_upload_app_store_connect` lane
 - Not called by neither the CI nor usually manually
 - Re-doing the same `ios_build_prechecks` and `ios_build_preflight` calls than its `build_and_upload_app_store_connect` called [already did](https://github.com/wordpress-mobile/WordPress-iOS/pull/18140/files#diff-72b7ec258b7b79c415642e3cb618b99665e0d8e491e4b8aafe468a88636c486fR83-R89)

So this is more like a private lane or helper… used by only one lane… which makes it a bit useless, and tbh is quite confusing as it exposes a new public lane to the user making it unclear if there's a difference between the 2 lanes when trying to figure out which one to call.

I'm thus suggesting that we cut/paste the body/implementation of this lane directly into `build_and_upload_app_store`, then remove this lane (called by no one else anyway), to avoid confusion between the 2 lanes. I've kept this idea for a separate PR though, to avoid making the diff and review of this PR harder than it already is, but hope to follow-up on it once this PR lands.

Reply:
Addressed in https://github.com/wordpress-mobile/WordPress-iOS/pull/18192